src/lib/reactOptimization.ts

```
@@ -157,9 +157,11 @@ export function memoize<T extends (...args: any[]) => any>(
 export function useDeepMemo<T>(value: T): T {
   const ref = useRef<T>(value);
   
-  if (!deepEqual(ref.current, value)) {
+   
+    // eslint-disable-next-line react-hooks/refs
     ref.current = value;
-  }
+  // eslint-disable-next-line react-hooks/refs
+  return ref.current;
   
   return ref.current;
```

Biome (2.1.2)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)